router/router.go

```diff
@@ -2306,6 +2305,9 @@ func (rt *HandleT) Start() {
 	ctx := rt.backgroundCtx
 	rt.backgroundGroup.Go(func() error {
 		<-rt.backendConfigInitialized
+		if rt.customDestinationManager != nil {
+			<-rt.customDestinationManager.BackendConfigInitialized()
```

Comment:
Given that we got a `ctx` here, wouldn't it make sense to `select` here to avoid any possible leak?